src/app/forms/custom-sections/plates/plates.component.ts

```diff
@@ -0,0 +1,110 @@
+import { HttpEventType } from '@angular/common/http';
+import { Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DocumentRetrievalService } from '@api/document-retrieval';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode, FormNodeEditTypes } from '@forms/services/dynamic-form.types';
+import { PlatesTemplate } from '@forms/templates/general/plates.template';
+import { Roles } from '@models/roles.enum';
+import { Plates, TechRecordModel } from '@models/vehicle-tech-record.model';
+import cloneDeep from 'lodash.clonedeep';
+import { debounceTime, Subscription, takeWhile } from 'rxjs';
+
+@Component({
+  selector: 'app-plates',
+  templateUrl: './plates.component.html',
+  styleUrls: ['./plates.component.scss']
+})
+export class PlatesComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() vehicleTechRecord!: TechRecordModel;
+  @Input() isEditing = false;
+
+  @Output() formChange = new EventEmitter();
+  @Output() isSuccess = new EventEmitter<boolean>();
+
+  public form!: CustomFormGroup;
+  private _formSubscription = new Subscription();
+  public isError: boolean = false;
+  public errorMessage?: string;
+
+  constructor(
+    public dfs: DynamicFormService,
+    private documentRetrievalService: DocumentRetrievalService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.form = this.dfs.createForm(PlatesTemplate, this.vehicleTechRecord) as CustomFormGroup;
+    this._formSubscription = this.form.cleanValueChanges.pipe(debounceTime(400)).subscribe(event => this.formChange.emit(event));
+  }
+
+  ngOnChanges(): void {
+    this.form?.patchValue(this.vehicleTechRecord, { emitEvent: false });
+  }
+
+  ngOnDestroy(): void {
+    this._formSubscription.unsubscribe();
+  }
+
+  hasPlates(): boolean {
+    return this.vehicleTechRecord.plates !== undefined && this.vehicleTechRecord.plates.length > 0;
+  }
+
+  get types(): typeof FormNodeEditTypes {
+    return FormNodeEditTypes;
+  }
+
+  get roles(): typeof Roles {
+    return Roles;
+  }
+
+  get mostRecentPlate(): Plates | undefined {
+    const platesList = cloneDeep(this.vehicleTechRecord.plates);
+    return platesList
+      ?.sort((a, b) => (a.plateIssueDate && b.plateIssueDate ? new Date(a.plateIssueDate).getTime() - new Date(b.plateIssueDate).getTime() : 0))
+      ?.pop();
```

Comment:
This is great, but your can return straight away without storing the clone in a variable.
```suggestion
    return cloneDeep(this.vehicleTechRecord.plates)
      ?.sort((a, b) => (a.plateIssueDate && b.plateIssueDate ? new Date(a.plateIssueDate).getTime() - new Date(b.plateIssueDate).getTime() : 0))
      ?.pop();
```